sql/core/src/main/scala/org/apache/spark/sql/execution/Window.scala

```diff
@@ -937,6 +945,12 @@ private[execution] object AggregateProcessor {
       imperatives.toArray,
       trackPartitionSize)
   }
+
+  // SPARK-14244: When binding
+  def windowPartitionSizeAttribute(expressions: Array[Expression]): AttributeReference = {
+    val aggs = expressions.flatMap(_.collectFirst { case f: SizeBasedWindowFunction => f })
+    aggs.headOption.map(_.n).getOrElse(SizeBasedWindowFunction.n)
```

Comment:
If we do not have any `SizeBasedWindowFunction`, should this method be called? When will we need to use `SizeBasedWindowFunction.n` created at the executor side? I guess I am confused why we need https://github.com/apache/spark/pull/12040/files#diff-5dade93a3cce85de2ceeab02bd9d4ff6R931.